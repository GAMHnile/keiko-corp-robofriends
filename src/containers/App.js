import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';

import './App.css';



class App extends Component {
  
  componentDidMount() {
    this.props.onRequestRobots();
  }

  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props;
    const filteredRobots = robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    })
    return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1 id="loading-robots">Loading</h1> :
            <ErrorBoundry>
              <CardList robots={filteredRobots} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

export default App;
