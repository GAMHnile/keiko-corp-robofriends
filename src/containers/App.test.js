import React from "react";
import App from "./App";
import { shallow } from "enzyme";

describe("App redux connected component test", () => {
  let AppWrapper;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      robots: [
        {
          id: 1,
          name: "John snow",
          email: "john@SpeechGrammarList.com",
        },
        {
          id: 2,
          name: "jane snow",
          email: "jane@SpeechGrammarList.com",
        },
        {
          id: 3,
          name: "Chris snow",
          email: "chris@SpeechGrammarList.com",
        },
      ],
      onRequestRobots: jest.fn(),
      searchField: "",
    };
    AppWrapper = shallow(<App {...mockProps} />)
  });
  it("App snapshot test", () => {
    expect(AppWrapper).toMatchSnapshot();
  });

  it("App shows loading when result ispending", () => {
    const AppWrapper = shallow(
      <App isPending={true} {...mockProps} robots={[]} />
    );
    expect(AppWrapper.find("[id='loading-robots']").length).toEqual(1);
  });
});
