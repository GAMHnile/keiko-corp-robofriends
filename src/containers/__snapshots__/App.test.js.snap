// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App redux connected component test App snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    onRequestRobots={[Function]}
    robots={
        Array [
            Object {
              "email": "john@SpeechGrammarList.com",
              "id": 1,
              "name": "John snow",
            },
            Object {
              "email": "jane@SpeechGrammarList.com",
              "id": 2,
              "name": "jane snow",
            },
            Object {
              "email": "chris@SpeechGrammarList.com",
              "id": 3,
              "name": "Chris snow",
            },
          ]
    }
    searchField=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SearchBox
          searchChange={undefined}
/>,
        <Scroll>
          <ErrorBoundary>
                    <CardList
                              robots={
                                        Array [
                                                  Object {
                                                    "email": "john@SpeechGrammarList.com",
                                                    "id": 1,
                                                    "name": "John snow",
                                                  },
                                                  Object {
                                                    "email": "jane@SpeechGrammarList.com",
                                                    "id": 2,
                                                    "name": "jane snow",
                                                  },
                                                  Object {
                                                    "email": "chris@SpeechGrammarList.com",
                                                    "id": 3,
                                                    "name": "Chris snow",
                                                  },
                                                ]
                              }
                    />
          </ErrorBoundary>
</Scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchChange": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundary>
            <CardList
                        robots={
                                    Array [
                                                Object {
                                                  "email": "john@SpeechGrammarList.com",
                                                  "id": 1,
                                                  "name": "John snow",
                                                },
                                                Object {
                                                  "email": "jane@SpeechGrammarList.com",
                                                  "id": 2,
                                                  "name": "jane snow",
                                                },
                                                Object {
                                                  "email": "chris@SpeechGrammarList.com",
                                                  "id": 3,
                                                  "name": "Chris snow",
                                                },
                                              ]
                        }
            />
</ErrorBoundary>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CardList
              robots={
                            Array [
                                          Object {
                                            "email": "john@SpeechGrammarList.com",
                                            "id": 1,
                                            "name": "John snow",
                                          },
                                          Object {
                                            "email": "jane@SpeechGrammarList.com",
                                            "id": 2,
                                            "name": "jane snow",
                                          },
                                          Object {
                                            "email": "chris@SpeechGrammarList.com",
                                            "id": 3,
                                            "name": "Chris snow",
                                          },
                                        ]
              }
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [
                Object {
                  "email": "john@SpeechGrammarList.com",
                  "id": 1,
                  "name": "John snow",
                },
                Object {
                  "email": "jane@SpeechGrammarList.com",
                  "id": 2,
                  "name": "jane snow",
                },
                Object {
                  "email": "chris@SpeechGrammarList.com",
                  "id": 3,
                  "name": "Chris snow",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SearchBox
            searchChange={undefined}
/>,
          <Scroll>
            <ErrorBoundary>
                        <CardList
                                    robots={
                                                Array [
                                                            Object {
                                                              "email": "john@SpeechGrammarList.com",
                                                              "id": 1,
                                                              "name": "John snow",
                                                            },
                                                            Object {
                                                              "email": "jane@SpeechGrammarList.com",
                                                              "id": 2,
                                                              "name": "jane snow",
                                                            },
                                                            Object {
                                                              "email": "chris@SpeechGrammarList.com",
                                                              "id": 3,
                                                              "name": "Chris snow",
                                                            },
                                                          ]
                                    }
                        />
            </ErrorBoundary>
</Scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchChange": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundary>
              <CardList
                            robots={
                                          Array [
                                                        Object {
                                                          "email": "john@SpeechGrammarList.com",
                                                          "id": 1,
                                                          "name": "John snow",
                                                        },
                                                        Object {
                                                          "email": "jane@SpeechGrammarList.com",
                                                          "id": 2,
                                                          "name": "jane snow",
                                                        },
                                                        Object {
                                                          "email": "chris@SpeechGrammarList.com",
                                                          "id": 3,
                                                          "name": "Chris snow",
                                                        },
                                                      ]
                            }
              />
</ErrorBoundary>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <CardList
                robots={
                                Array [
                                                Object {
                                                  "email": "john@SpeechGrammarList.com",
                                                  "id": 1,
                                                  "name": "John snow",
                                                },
                                                Object {
                                                  "email": "jane@SpeechGrammarList.com",
                                                  "id": 2,
                                                  "name": "jane snow",
                                                },
                                                Object {
                                                  "email": "chris@SpeechGrammarList.com",
                                                  "id": 3,
                                                  "name": "Chris snow",
                                                },
                                              ]
                }
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [
                  Object {
                    "email": "john@SpeechGrammarList.com",
                    "id": 1,
                    "name": "John snow",
                  },
                  Object {
                    "email": "jane@SpeechGrammarList.com",
                    "id": 2,
                    "name": "jane snow",
                  },
                  Object {
                    "email": "chris@SpeechGrammarList.com",
                    "id": 3,
                    "name": "Chris snow",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
